src/plugins/discover/public/application/main/utils/get_data_view_by_text_based_query_lang.ts

```diff
@@ -25,6 +25,9 @@ export async function getDataViewByTextBasedQueryLang(
   }
   if ('esql' in query) {
     indexPatternFromQuery = getIndexPatternFromESQLQuery(query.esql);
+    if (indexPatternFromQuery === '') {
+      return new DataView({ fieldFormats: services.fieldFormats });
+    }
```

Comment:
Looking good from my end, question is, if this could be further optimized, we're saving redundant settings here 
![Bildschirmfoto 2024-03-21 um 18 57 27](https://github.com/elastic/kibana/assets/463851/030f9c55-9400-4b07-8666-ea28f197c71b)

neither title, nor sourceFilters, nor fieldFormats, nor runtimeFieldMap, ...
would be necessary to be persisted as serialized values, since those are just defaults. just persisting what's necessary, would also make it easier to debug, and save a bit of space (not much, but well, a tiny little bit)

Reply:
https://github.com/elastic/kibana/issues/180429